components/interactive_dialog/dialog_element/index.ts

```diff
@@ -7,17 +7,15 @@ import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
 import {autocompleteChannels} from 'actions/channel_actions';
 import {autocompleteUsers} from 'actions/user_actions';
 
-import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
-
-import {UserProfile} from '@mattermost/types/users';
+import {ActionFunc, DispatchFunc, GenericAction} from 'mattermost-redux/types/actions';
 
 import {Channel} from '@mattermost/types/channels';
 
 import DialogElement from './dialog_element';
 
 type Actions = {
     autocompleteChannels: (term: string, success: (channels: Channel[]) => void, error: () => void) => Promise<void>;
-    autocompleteUsers: (search: string) => Promise<UserProfile[]>;
+    autocompleteUsers: (search: string) => (doDispatch: DispatchFunc) => Promise<any>;
```

Comment:
Same comment related to repetition of `Actions`. Just export from component and import & use it `index`